### DISCUSSION TITLE
Newbie, stuck during process

### DISCUSSION THREAD

**Discussion by longtry** [linux-surface/linux-surface#1158] [Q&A]
I have a SB2 I've followed your instructions successfully until these lines:

> However, if the default kernel has a higher version than the surface kernel it will override the surface kernel and get booted by default. You can use a systemd path unit to fix this:
Put this into /etc/systemd/system/default-kernel.path
[Unit]
Description=Fedora default kernel updater
[Path]
PathChanged=/boot
[Install]
WantedBy=default.target
And this into /etc/systemd/system/default-kernel.service
[Unit]
Description=Fedora default kernel updater
[Service]
Type=oneshot
ExecStart=/bin/sh -c "grubby --set-default /boot/vmlinuz*surface*"

I just can't create a file named "default-kernel.path" in such a folder in Fedora 38 using the default explorer app. Being a newbie, I read on the net and tried on the terminal with the command "cat", but still unsuccessful. Any tips?

**Comment by qzed** [comment]
Unfortunately, most (if not all) graphical explorer apps are notoriously bad at dealing with root / access to files of a different owner. So you usually need to do that sort of stuff via the terminal.

The key component for doing this is `sudo <command>`, which essentially executes `<command>` as root user, giving you access to protected files (or rather, files owned by root).  For example, you can combine `sudo` with the `nano` editor to create/edit root-owned files in the terminal:
```
sudo nano /etc/systemd/system/default-kernel.path
```

 Note that `sudo` may have problems with GUI apps (hence the use of `nano`).

**However**, in this case, I would recommend you have a look at https://github.com/linux-surface/linux-surface/tree/master/contrib/fedora-default-kernel. The instructions from the wiki were a simplified (and outdated) version of that. I've updated them to point at this directory now. In essence, you download the three `default-kernel` files from there and then you can just copy (`cp`) or move (`mv`) them to the proper location by invoking the command via `sudo`.

**Reply by qzed** [reply]
(and maybe some more explanation on the other commands: `chmod 755 <file>` marks `<file>` as executable, `systemctl daemon-reload` reloads systemctl daemon/service files after changes/additions, and `systemctl enable --now <service>` ensures that `<service>` is automatically started at boot, with `--now` stating that it should also be started right now)

**Reply by longtry** [reply]
Thanks. The 1st command went well. When I copied your 2nd line into terminal:
`sudo cp default-kernel.{path,service} /etc/systemd/system/`
It spat out

> cp: cannot stat 'default-kernel.path': No such file or directory
> cp: cannot stat 'default-kernel.service': No such file or directory

I thought maybe it's because I haven't created the file yet? So I went back and used your 1st answer's command
`sudo nano /etc/systemd/system/default-kernel.path`
to create 2 files, and double checked them to be there using the visuals from the "Files" app. Then I repeated the 2nd-line command, but the terminal still refused to proceed, displaying the same error as before. Is there something going on here?

**Reply by qzed** [reply]
The latest kernel packages on Fedora (6.3.6, see also https://github.com/linux-surface/linux-surface/issues/1159) now include that service. You should be able to just run `sudo systemctl enable --now linux-surface-default-kernel`.

**Comment by longtry** [comment]
A somewhat related question. Is Mint supported? I barely know that Mint is based on Ubuntu, but I'm not sure if I follow the Ubuntu instructions, Mint would work on my SB2.

**Reply by qzed** [reply]
It should work.

**Comment by longtry** [comment]
Another question. If I upgrade Linux Mint (in this case, from Vera to Victoria, the latest version), then would it still have our Surface "patch"? Or do I need to reinstall LS from scratch?

**Reply by qzed** [reply]
I don't know anything about Linux Mint, so I can't really say. I believe generally it should still be there. You can check and make sure that the repo is still present after updating and you can just follow the installation instructions again to make sure that the packages are installed. It should tell you if they are still there.

**Reply by longtry** [reply]
Thank you, qzed. Now I have a much more urgent problem of not being able to boot, as described in the [other thread](https://github.com/linux-surface/linux-surface/discussions/1194). Could you take a look and let me know what you think?
